docs/contributing/our-backward-compatibility-promise.md

```diff
@@ -0,0 +1,71 @@
+# Our Backward Compatibility Promise
+
+Smooth and safe upgrades of your own e-commerce project are very important to us.
+In the same time, we need to be able to improve Shopsys Framework for you by adding functionality, enhancing or simplifying current functions and fixing bugs.
+This is why we've written this page explaining what amount of backward compatibility should be expected and how we plan to make changes in the future.
+
+Shopsys Framework is built on the shoulders of giants so we've based our BC promise on the [Symfony Backward Compatibility Promise](https://symfony.com/doc/3.4/contributing/code/bc.html).
+Exceptions from adhering to Symfony's promise and clarifications for non-PHP source codes can be found below.
+
+*Note: This BC promise becomes effective with the release first non-beta release (`7.0.0`).*
+
+## Releases and Versioning
+This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html), which means it's release versions are in format `MAJOR.MINOR.PATCH`:
+
+- `MAJOR` version may contain incompatible changes
+- `MINOR` version may add new functionality in a backward-compatible manner
+- `PATCH` version contains only backward-compatible bug fixes
+
+Released versions will be always marked using git tags with `v` prefix (eg. `v7.0.0`).
+Once created, a git tag marking a release will never be removed or edited.
+
+*Note: Pre-release versions may introduce incompatible changes and can be used to try out the new functions and changes.
+Pre-release version format is `MAJOR.MINOR.PATCH-<alpha|beta|rc>n`, eg. `7.0.0-beta5`.*
+
+### Current Release Plan
+To be able to develop and improve Shopsys Framework we plan to release `MAJOR` versions almost quarterly, aiming to release a new `MAJOR` every 3-4 months.
+We expect this period to increase in the future to yearly releases.
+
+## Clarifications and Exceptions
+
+### Project-base Repository
+The [project-base repository](https://github.com/shopsys/project-base) is excepted for the BC promise because it is not meant to be depended upon or extended.
```

Comment:
I don't understand the sentence, please rephrase it for not so smart people like me.

Reply:
reworded in df1487112